src/coreComponents/mesh/generators/InternalWellGenerator.cpp

```diff
@@ -220,28 +223,45 @@ void InternalWellGenerator::constructPolylineNodeToSegmentMap()
     real64 const segmentLength = LvArray::tensorOps::l2Norm< 3 >( vSeg );
 
     // various checks and warnings on the segment and element length
-    GEOSX_THROW_IF( segmentLength < 1e-2,
-                    "Error in the topology of well " << getName() <<
-                    ": we detected a polyline segment measuring less than 1 cm",
+    GEOSX_THROW_IF( segmentLength < 1e-5,
+                    "Error in the topology of well '" << getName() <<
+                    "': we detected a polyline segment measuring less than 1 cm",
+                    InputError );
+
+    GEOSX_THROW_IF( m_polyNodeCoords[ipolyNode_a][2] < m_polyNodeCoords[ipolyNode_b][2],
+                    "Error in the topology of well '" << getName() <<
+                    "': in the polyline, each segment must be going down. \n" <<
+                    "This is not the case between polyline nodes " << m_polyNodeCoords[ipolyNode_a] << " and " << m_polyNodeCoords[ipolyNode_b],
                     InputError );
 
-    if( segmentLength < 1.0 )
+    if( segmentLength < 0.1 )
     {
-      GEOSX_LOG_RANK_0(
-        "\n \nWarning: we detected a segment measuring less than 1 m in the topology of well " << getName() << "\n"
-                                                                                               << "The simulation can proceed like that, but very small segments may cause numerical issues, so it is something to keep an eye on.\n \n" );
+      foundSmallSegment = true;
     }
 
     if( segmentLength / m_numElemsPerSegment < 1e-2 )
     {
-      GEOSX_LOG_RANK_0( "\n \nWarning: the chosen number of elements per polyline segment (" << m_numElemsPerSegment <<
-                        ") leads to well elements measuring less than 1 cm in the topology of well " << getName() << "\n \n" );
+      foundSmallWellElement = true;
     }
 
     // map the polyline node ids to the polyline segment ids
     m_polyNodeToSegmentMap[ipolyNode_a].insert( iseg );
     m_polyNodeToSegmentMap[ipolyNode_b].insert( iseg );
   }
+
+  if( foundSmallSegment )
+  {
+    GEOSX_LOG_RANK_0(
+      "\nWarning: we detected a segment measuring less than 1 m in the topology of well '" << getName() << "'\n"
```

Comment:
Same, please update the error message accordingly (`1 m` -> `minSegmentLength m`).

Reply:
I updated the error message